In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv')

In [3]:
df.head(4)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S

In [4]:
df.tail(3)

PassengerId  Survived  Pclass                                      Name  \
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

In [5]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [6]:
df.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

In [7]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [8]:
df.dtypes == 'object'

PassengerId    False
Survived       False
Pclass         False
Name            True
Sex             True
Age            False
SibSp          False
Parch          False
Ticket          True
Fare           False
Cabin           True
Embarked        True
dtype: bool

In [9]:
type(df.dtypes == 'object')

pandas.core.series.Series

In [10]:
df.dtypes[df.dtypes == 'object'].index

Index(['Name', 'Sex', 'Ticket', 'Cabin', 'Embarked'], dtype='object')

In [11]:
col_name = df.dtypes[df.dtypes == 'object'].index

In [12]:
df[col_name].describe()

Name   Sex  Ticket    Cabin Embarked
count                       891   891     891      204      889
unique                      891     2     681      147        3
top     Braund, Mr. Owen Harris  male  347082  B96 B98        S
freq                          1   577       7        4      644

In [13]:
df.head(3)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S

In [14]:
df['Hero'] = "WOW"

In [15]:
df.head(1)

PassengerId  Survived  Pclass                     Name   Sex   Age  SibSp  \
0            1         0       3  Braund, Mr. Owen Harris  male  22.0      1   

   Parch     Ticket  Fare Cabin Embarked Hero  
0      0  A/5 21171  7.25   NaN        S  WOW

In [16]:
df['Name'][0:5]

0                              Braund, Mr. Owen Harris
1    Cumings, Mrs. John Bradley (Florence Briggs Th...
2                               Heikkinen, Miss. Laina
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)
4                             Allen, Mr. William Henry
Name: Name, dtype: object

In [17]:
df['Name'][0:15:3]

0                          Braund, Mr. Owen Harris
3     Futrelle, Mrs. Jacques Heath (Lily May Peel)
6                          McCarthy, Mr. Timothy J
9              Nasser, Mrs. Nicholas (Adele Achem)
12                  Saundercock, Mr. William Henry
Name: Name, dtype: object

In [18]:
df[df['Age'].isnull() == True].index

Int64Index([  5,  17,  19,  26,  28,  29,  31,  32,  36,  42,
            ...
            832, 837, 839, 846, 849, 859, 863, 868, 878, 888],
           dtype='int64', length=177)

In [19]:
len(df[df['Age'].isnull() == True].index)

177

In [20]:
ind = df[df['Age'].isnull() == True].index

In [21]:
df.loc[ind]

PassengerId  Survived  Pclass                                      Name  \
5              6         0       3                          Moran, Mr. James   
17            18         1       2              Williams, Mr. Charles Eugene   
19            20         1       3                   Masselmani, Mrs. Fatima   
26            27         0       3                   Emir, Mr. Farred Chehab   
28            29         1       3             O'Dwyer, Miss. Ellen "Nellie"   
..           ...       ...     ...                                       ...   
859          860         0       3                          Razi, Mr. Raihed   
863          864         0       3         Sage, Miss. Dorothy Edith "Dolly"   
868          869         0       3               van Melkebeke, Mr. Philemon   
878          879         0       3                        Laleff, Mr. Kristo   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   

        Sex  Age  SibSp  Parch      Ticket     Fare Cabin Embarked Hero  
5      male  NaN      0      0      330877   8.4583   NaN        Q  WOW  
17     male  NaN      0      0      244373  13.0000   NaN        S  WOW  
19   female  NaN      0      0        2649   7.2250   NaN        C  WOW  
26     male  NaN      0      0        2631   7.2250   NaN        C  WOW  
28   female  NaN      0      0      330959   7.8792   NaN        Q  WOW  
..      ...  ...    ...    ...         ...      ...   ...      ...  ...  
859    male  NaN      0      0        2629   7.2292   NaN        C  WOW  
863  female  NaN      8      2    CA. 2343  69.5500   NaN        S  WOW  
868    male  NaN      0      0      345777   9.5000   NaN        S  WOW  
878    male  NaN      0      0      349217   7.8958   NaN        S  WOW  
888  female  NaN      1      2  W./C. 6607  23.4500   NaN        S  WOW  

[177 rows x 13 columns]

In [22]:
df[df['Age'].isnull()]

PassengerId  Survived  Pclass                                      Name  \
5              6         0       3                          Moran, Mr. James   
17            18         1       2              Williams, Mr. Charles Eugene   
19            20         1       3                   Masselmani, Mrs. Fatima   
26            27         0       3                   Emir, Mr. Farred Chehab   
28            29         1       3             O'Dwyer, Miss. Ellen "Nellie"   
..           ...       ...     ...                                       ...   
859          860         0       3                          Razi, Mr. Raihed   
863          864         0       3         Sage, Miss. Dorothy Edith "Dolly"   
868          869         0       3               van Melkebeke, Mr. Philemon   
878          879         0       3                        Laleff, Mr. Kristo   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   

        Sex  Age  SibSp  Parch      Ticket     Fare Cabin Embarked Hero  
5      male  NaN      0      0      330877   8.4583   NaN        Q  WOW  
17     male  NaN      0      0      244373  13.0000   NaN        S  WOW  
19   female  NaN      0      0        2649   7.2250   NaN        C  WOW  
26     male  NaN      0      0        2631   7.2250   NaN        C  WOW  
28   female  NaN      0      0      330959   7.8792   NaN        Q  WOW  
..      ...  ...    ...    ...         ...      ...   ...      ...  ...  
859    male  NaN      0      0        2629   7.2292   NaN        C  WOW  
863  female  NaN      8      2    CA. 2343  69.5500   NaN        S  WOW  
868    male  NaN      0      0      345777   9.5000   NaN        S  WOW  
878    male  NaN      0      0      349217   7.8958   NaN        S  WOW  
888  female  NaN      1      2  W./C. 6607  23.4500   NaN        S  WOW  

[177 rows x 13 columns]

In [23]:
max(df['Fare'])

512.3292

In [24]:
df[df['Fare'] == max(df['Fare'])]

PassengerId  Survived  Pclass                                Name  \
258          259         1       1                    Ward, Miss. Anna   
679          680         1       1  Cardeza, Mr. Thomas Drake Martinez   
737          738         1       1              Lesurer, Mr. Gustave J   

        Sex   Age  SibSp  Parch    Ticket      Fare        Cabin Embarked Hero  
258  female  35.0      0      0  PC 17755  512.3292          NaN        C  WOW  
679    male  36.0      0      1  PC 17755  512.3292  B51 B53 B55        C  WOW  
737    male  35.0      0      0  PC 17755  512.3292         B101        C  WOW

##### Challenges

1. Find out how many male and female passenger was onboarded .
2. how many survived?
3. how many casuality? 
4. what is name of a person who is the eldest one . 
5. how many passenger do we have in first , second and third class 
6. how many person we have whose name starts with "s"
7. try to create a new column which is a summation  of "SibSp" and "parch"
8. how many person do we have below age of 25?
9. how many person died whose age was less then 40 
10. from a  cabin column seperate text and numeric value . 


#### 1. Find out how many male and female passenger was onboarded .

In [25]:
df[df['Sex']== "male"]

PassengerId  Survived  Pclass                            Name   Sex  \
0              1         0       3         Braund, Mr. Owen Harris  male   
4              5         0       3        Allen, Mr. William Henry  male   
5              6         0       3                Moran, Mr. James  male   
6              7         0       1         McCarthy, Mr. Timothy J  male   
7              8         0       3  Palsson, Master. Gosta Leonard  male   
..           ...       ...     ...                             ...   ...   
883          884         0       2   Banfield, Mr. Frederick James  male   
884          885         0       3          Sutehall, Mr. Henry Jr  male   
886          887         0       2           Montvila, Rev. Juozas  male   
889          890         1       1           Behr, Mr. Karl Howell  male   
890          891         0       3             Dooley, Mr. Patrick  male   

      Age  SibSp  Parch            Ticket     Fare Cabin Embarked Hero  
0    22.0      1      0         A/5 21171   7.2500   NaN        S  WOW  
4    35.0      0      0            373450   8.0500   NaN        S  WOW  
5     NaN      0      0            330877   8.4583   NaN        Q  WOW  
6    54.0      0      0             17463  51.8625   E46        S  WOW  
7     2.0      3      1            349909  21.0750   NaN        S  WOW  
..    ...    ...    ...               ...      ...   ...      ...  ...  
883  28.0      0      0  C.A./SOTON 34068  10.5000   NaN        S  WOW  
884  25.0      0      0   SOTON/OQ 392076   7.0500   NaN        S  WOW  
886  27.0      0      0            211536  13.0000   NaN        S  WOW  
889  26.0      0      0            111369  30.0000  C148        C  WOW  
890  32.0      0      0            370376   7.7500   NaN        Q  WOW  

[577 rows x 13 columns]

In [26]:
df[df['Sex']== "female"]

PassengerId  Survived  Pclass  \
1              2         1       1   
2              3         1       3   
3              4         1       1   
8              9         1       3   
9             10         1       2   
..           ...       ...     ...   
880          881         1       2   
882          883         0       3   
885          886         0       3   
887          888         1       1   
888          889         0       3   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
..                                                 ...     ...   ...    ...   
880       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
882                       Dahlberg, Miss. Gerda Ulrika  female  22.0      0   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   

     Parch            Ticket     Fare Cabin Embarked Hero  
1        0          PC 17599  71.2833   C85        C  WOW  
2        0  STON/O2. 3101282   7.9250   NaN        S  WOW  
3        0            113803  53.1000  C123        S  WOW  
8        2            347742  11.1333   NaN        S  WOW  
9        0            237736  30.0708   NaN        C  WOW  
..     ...               ...      ...   ...      ...  ...  
880      1            230433  26.0000   NaN        S  WOW  
882      0              7552  10.5167   NaN        S  WOW  
885      5            382652  29.1250   NaN        Q  WOW  
887      0            112053  30.0000   B42        S  WOW  
888      2        W./C. 6607  23.4500   NaN        S  WOW  

[314 rows x 13 columns]

In [27]:
df['Sex'].value_counts()

male      577
female    314
Name: Sex, dtype: int64

In [28]:
df.groupby('Sex').count()

PassengerId  Survived  Pclass  Name  Age  SibSp  Parch  Ticket  Fare  \
Sex                                                                            
female          314       314     314   314  261    314    314     314   314   
male            577       577     577   577  453    577    577     577   577   

        Cabin  Embarked  Hero  
Sex                            
female     97       312   314  
male      107       577   577

In [29]:
df.groupby('Sex')['Sex'].count()

Sex
female    314
male      577
Name: Sex, dtype: int64

##### 577 male and 314 female passengers

#### 2. how many survived?


In [30]:
len(df[df['Survived']== True])

342

##### 342 passengers survived

#### 3. how many casuality? 

In [31]:
len(df[df['Survived']== False])

549

In [32]:
df['Survived'].value_counts()

0    549
1    342
Name: Survived, dtype: int64

#### 549 casualties

#### 4. what is name of a person who is the eldest one . 


In [33]:
df[df['Age']== max(df['Age'])]


PassengerId  Survived  Pclass                                  Name  \
630          631         1       1  Barkworth, Mr. Algernon Henry Wilson   

      Sex   Age  SibSp  Parch Ticket  Fare Cabin Embarked Hero  
630  male  80.0      0      0  27042  30.0   A23        S  WOW

In [34]:
df[df['Age']== max(df['Age'])]['Name']

630    Barkworth, Mr. Algernon Henry Wilson
Name: Name, dtype: object

##### Mr Algernon

##### 5. how many passenger do we have in first , second and third class 


In [35]:
# 491 in pclass 3
df[df['Pclass']==3]

PassengerId  Survived  Pclass                                      Name  \
0              1         0       3                   Braund, Mr. Owen Harris   
2              3         1       3                    Heikkinen, Miss. Laina   
4              5         0       3                  Allen, Mr. William Henry   
5              6         0       3                          Moran, Mr. James   
7              8         0       3            Palsson, Master. Gosta Leonard   
..           ...       ...     ...                                       ...   
882          883         0       3              Dahlberg, Miss. Gerda Ulrika   
884          885         0       3                    Sutehall, Mr. Henry Jr   
885          886         0       3      Rice, Mrs. William (Margaret Norton)   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch            Ticket     Fare Cabin Embarked Hero  
0      male  22.0      1      0         A/5 21171   7.2500   NaN        S  WOW  
2    female  26.0      0      0  STON/O2. 3101282   7.9250   NaN        S  WOW  
4      male  35.0      0      0            373450   8.0500   NaN        S  WOW  
5      male   NaN      0      0            330877   8.4583   NaN        Q  WOW  
7      male   2.0      3      1            349909  21.0750   NaN        S  WOW  
..      ...   ...    ...    ...               ...      ...   ...      ...  ...  
882  female  22.0      0      0              7552  10.5167   NaN        S  WOW  
884    male  25.0      0      0   SOTON/OQ 392076   7.0500   NaN        S  WOW  
885  female  39.0      0      5            382652  29.1250   NaN        Q  WOW  
888  female   NaN      1      2        W./C. 6607  23.4500   NaN        S  WOW  
890    male  32.0      0      0            370376   7.7500   NaN        Q  WOW  

[491 rows x 13 columns]

In [36]:
len(df[df['Pclass']==1])

216

In [37]:
len(df[df['Pclass']==2])

184

In [38]:
df['Pclass'].value_counts()

3    491
1    216
2    184
Name: Pclass, dtype: int64

In [39]:
df.groupby('Pclass')['Pclass'].count()

Pclass
1    216
2    184
3    491
Name: Pclass, dtype: int64

Pclass 1 = 216

Pclass 2 = 184

Pclass 3 = 491

##### 6. how many person we have whose name starts with "S"


In [40]:
df[df['Name'].str.startswith('S')]

PassengerId  Survived  Pclass  \
10            11         1       3   
12            13         0       3   
23            24         1       1   
31            32         1       1   
48            49         0       3   
..           ...       ...     ...   
851          852         0       3   
862          863         1       1   
863          864         0       3   
880          881         1       2   
884          885         0       3   

                                                  Name     Sex   Age  SibSp  \
10                     Sandstrom, Miss. Marguerite Rut  female   4.0      1   
12                      Saundercock, Mr. William Henry    male  20.0      0   
23                        Sloper, Mr. William Thompson    male  28.0      0   
31      Spencer, Mrs. William Augustus (Marie Eugenie)  female   NaN      1   
48                                 Samaan, Mr. Youssef    male   NaN      2   
..                                                 ...     ...   ...    ...   
851                                Svensson, Mr. Johan    male  74.0      0   
862  Swift, Mrs. Frederick Joel (Margaret Welles Ba...  female  48.0      0   
863                  Sage, Miss. Dorothy Edith "Dolly"  female   NaN      8   
880       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
884                             Sutehall, Mr. Henry Jr    male  25.0      0   

     Parch           Ticket      Fare Cabin Embarked Hero  
10       1          PP 9549   16.7000    G6        S  WOW  
12       0        A/5. 2151    8.0500   NaN        S  WOW  
23       0           113788   35.5000    A6        S  WOW  
31       0         PC 17569  146.5208   B78        C  WOW  
48       0             2662   21.6792   NaN        C  WOW  
..     ...              ...       ...   ...      ...  ...  
851      0           347060    7.7750   NaN        S  WOW  
862      0            17466   25.9292   D17        S  WOW  
863      2         CA. 2343   69.5500   NaN        S  WOW  
880      1           230433   26.0000   NaN        S  WOW  
884      0  SOTON/OQ 392076    7.0500   NaN        S  WOW  

[86 rows x 13 columns]

In [41]:
len(df[df['Name'].str.startswith('S')]['Name'])

86

##### No name starts with s

#### 7. try to create a new column which is a summation  of "SibSp" and "parch"


In [42]:
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked Hero  
0        0         A/5 21171   7.2500   NaN        S  WOW  
1        0          PC 17599  71.2833   C85        C  WOW  
2        0  STON/O2. 3101282   7.9250   NaN        S  WOW  
3        0            113803  53.1000  C123        S  WOW  
4        0            373450   8.0500   NaN        S  WOW  
..     ...               ...      ...   ...      ...  ...  
886      0            211536  13.0000   NaN        S  WOW  
887      0            112053  30.0000   B42        S  WOW  
888      2        W./C. 6607  23.4500   NaN        S  WOW  
889      0            111369  30.0000  C148        C  WOW  
890      0            370376   7.7500   NaN        Q  WOW  

[891 rows x 13 columns]

In [43]:
summ= df['SibSp'] + df['Parch']

In [44]:
df['New_total']= summ

In [45]:
df.head(3)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   

   Parch            Ticket     Fare Cabin Embarked Hero  New_total  
0      0         A/5 21171   7.2500   NaN        S  WOW          1  
1      0          PC 17599  71.2833   C85        C  WOW          1  
2      0  STON/O2. 3101282   7.9250   NaN        S  WOW          0

8. how many person do we have below age of 25?

In [46]:
len(df[df['Age']<25])

278

278 below the age of 25

##### 9. how many person died whose age was less then 40 


In [47]:
df[df['Age']<40]

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
885          886         0       3   
886          887         0       2   
887          888         1       1   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked Hero  New_total  
0        0         A/5 21171   7.2500   NaN        S  WOW          1  
1        0          PC 17599  71.2833   C85        C  WOW          1  
2        0  STON/O2. 3101282   7.9250   NaN        S  WOW          0  
3        0            113803  53.1000  C123        S  WOW          1  
4        0            373450   8.0500   NaN        S  WOW          0  
..     ...               ...      ...   ...      ...  ...        ...  
885      5            382652  29.1250   NaN        Q  WOW          5  
886      0            211536  13.0000   NaN        S  WOW          0  
887      0            112053  30.0000   B42        S  WOW          0  
889      0            111369  30.0000  C148        C  WOW          0  
890      0            370376   7.7500   NaN        Q  WOW          0  

[551 rows x 14 columns]

In [48]:
df[df['Survived']== 0]

PassengerId  Survived  Pclass                                      Name  \
0              1         0       3                   Braund, Mr. Owen Harris   
4              5         0       3                  Allen, Mr. William Henry   
5              6         0       3                          Moran, Mr. James   
6              7         0       1                   McCarthy, Mr. Timothy J   
7              8         0       3            Palsson, Master. Gosta Leonard   
..           ...       ...     ...                                       ...   
884          885         0       3                    Sutehall, Mr. Henry Jr   
885          886         0       3      Rice, Mrs. William (Margaret Norton)   
886          887         0       2                     Montvila, Rev. Juozas   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch           Ticket     Fare Cabin Embarked Hero  \
0      male  22.0      1      0        A/5 21171   7.2500   NaN        S  WOW   
4      male  35.0      0      0           373450   8.0500   NaN        S  WOW   
5      male   NaN      0      0           330877   8.4583   NaN        Q  WOW   
6      male  54.0      0      0            17463  51.8625   E46        S  WOW   
7      male   2.0      3      1           349909  21.0750   NaN        S  WOW   
..      ...   ...    ...    ...              ...      ...   ...      ...  ...   
884    male  25.0      0      0  SOTON/OQ 392076   7.0500   NaN        S  WOW   
885  female  39.0      0      5           382652  29.1250   NaN        Q  WOW   
886    male  27.0      0      0           211536  13.0000   NaN        S  WOW   
888  female   NaN      1      2       W./C. 6607  23.4500   NaN        S  WOW   
890    male  32.0      0      0           370376   7.7500   NaN        Q  WOW   

     New_total  
0            1  
4            0  
5            0  
6            0  
7            4  
..         ...  
884          0  
885          5  
886          0  
888          3  
890          0  

[549 rows x 14 columns]

In [49]:
df[(df['Age']<40) & (df['Survived']== 0)]


PassengerId  Survived  Pclass                                  Name  \
0              1         0       3               Braund, Mr. Owen Harris   
4              5         0       3              Allen, Mr. William Henry   
7              8         0       3        Palsson, Master. Gosta Leonard   
12            13         0       3        Saundercock, Mr. William Henry   
13            14         0       3           Andersson, Mr. Anders Johan   
..           ...       ...     ...                                   ...   
883          884         0       2         Banfield, Mr. Frederick James   
884          885         0       3                Sutehall, Mr. Henry Jr   
885          886         0       3  Rice, Mrs. William (Margaret Norton)   
886          887         0       2                 Montvila, Rev. Juozas   
890          891         0       3                   Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch            Ticket    Fare Cabin Embarked Hero  \
0      male  22.0      1      0         A/5 21171   7.250   NaN        S  WOW   
4      male  35.0      0      0            373450   8.050   NaN        S  WOW   
7      male   2.0      3      1            349909  21.075   NaN        S  WOW   
12     male  20.0      0      0         A/5. 2151   8.050   NaN        S  WOW   
13     male  39.0      1      5            347082  31.275   NaN        S  WOW   
..      ...   ...    ...    ...               ...     ...   ...      ...  ...   
883    male  28.0      0      0  C.A./SOTON 34068  10.500   NaN        S  WOW   
884    male  25.0      0      0   SOTON/OQ 392076   7.050   NaN        S  WOW   
885  female  39.0      0      5            382652  29.125   NaN        Q  WOW   
886    male  27.0      0      0            211536  13.000   NaN        S  WOW   
890    male  32.0      0      0            370376   7.750   NaN        Q  WOW   

     New_total  
0            1  
4            0  
7            4  
12           0  
13           6  
..         ...  
883          0  
884          0  
885          5  
886          0  
890          0  

[322 rows x 14 columns]

##### 322 passengers of age less than 40 passed away

10. from a  cabin column seperate text and numeric value . 

In [52]:
df.head(2)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   

   Parch     Ticket     Fare Cabin Embarked Hero  New_total  
0      0  A/5 21171   7.2500   NaN        S  WOW          1  
1      0   PC 17599  71.2833   C85        C  WOW          1

In [53]:
import re

In [64]:
df['cabin_letter']= df['Cabin'].str.extract('([a-zA-Z]+)')
df['cabin_num']= df['Cabin'].str.extract('([^a-zA-Z]+)')

In [65]:
df.head(2)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   

   Parch     Ticket     Fare Cabin Embarked Hero  New_total cabin_letter  \
0      0  A/5 21171   7.2500   NaN        S  WOW          1          NaN   
1      0   PC 17599  71.2833   C85        C  WOW          1            C   

  cabin_num  
0       NaN  
1        85